### ISSUE TITLE
GenModel publicationLocation does not work with Xcore

### ISSUE THREAD

**Issue by janhicken** [eclipse/xtext-maven#26]
Hi,

I'm using the annotation `@GenModel(publicationLocation="/my-project/output.ecore")` to produce an Ecore file from my Xcore model in the Maven build process. All the Java files are generated as usual, however, this exception gets printed to the console when executing the build:

```
java.io.IOException: The path '/my-project/output.ecore' is unmapped
	at org.eclipse.emf.ecore.resource.impl.PlatformResourceURIHandlerImpl.createOutputStream(PlatformResourceURIHandlerImpl.java:501)
	at org.eclipse.emf.ecore.resource.impl.ExtensibleURIConverterImpl.createOutputStream(ExtensibleURIConverterImpl.java:349)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.saveOnlyIfChangedWithMemoryBuffer(ResourceImpl.java:1234)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.save(ResourceImpl.java:985)
	at org.eclipse.emf.codegen.ecore.genmodel.generator.GenPackageGeneratorAdapter.generatePackagePublication(GenPackageGeneratorAdapter.java:358)
	at org.eclipse.emf.codegen.ecore.genmodel.generator.GenPackageGeneratorAdapter.generateModel(GenPackageGeneratorAdapter.java:212)
	at org.eclipse.emf.codegen.ecore.genmodel.generator.GenBaseGeneratorAdapter.doGenerate(GenBaseGeneratorAdapter.java:215)
	at org.eclipse.emf.codegen.ecore.generator.AbstractGeneratorAdapter.generate(AbstractGeneratorAdapter.java:317)
	at org.eclipse.emf.codegen.ecore.generator.Generator.generate(Generator.java:708)
	at org.eclipse.emf.codegen.ecore.generator.Generator.generate(Generator.java:619)
	at org.eclipse.emf.ecore.xcore.generator.XcoreGenerator.generateGenModel(XcoreGenerator.java:280)
	at org.eclipse.emf.ecore.xcore.generator.XcoreGenerator.doGenerate(XcoreGenerator.java:228)
	at org.eclipse.xtext.generator.GeneratorDelegate.doGenerate(GeneratorDelegate.java:45)
	at org.eclipse.xtext.generator.GeneratorDelegate.generate(GeneratorDelegate.java:34)
	at org.eclipse.xtext.builder.standalone.StandaloneBuilder.generate(StandaloneBuilder.java:406)
	at org.eclipse.xtext.builder.standalone.StandaloneBuilder.launch(StandaloneBuilder.java:279)
	at org.eclipse.xtext.maven.XtextGenerator.internalExecute(XtextGenerator.java:165)
	at org.eclipse.xtext.maven.XtextGenerator.execute(XtextGenerator.java:142)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:955)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:290)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:194)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```

It seems that the mapping at `EcorePlugin.getPlatformResourceMap()` hasn't been initialized so the platform URIs cannot be resolved against it. Is it possible to do so in a specific standlone setup mechanism?

MWE has similar stuff implemented in `org.eclipse.emf.mwe.utils.StandaloneSetup#addProjectMapping`

**Comment by cdietrich** [comment]
Xcore support with maven is general shitty

See https://github.com/cdietrich/xtext-ecore-xcore/blob/master/dully/src/dully/XcoreStandaloneSetup2.java

For a workaround to create such mapping’s

**Comment by janhicken** [comment]
I tried your workaround with my project setup and it seems to work fine.

However, the problem is, that the given Maven project directory structure is only accessible within the Mojo, so I can only hard-code the entries to fill into the platform resource map.

**Comment by holgerschill** [comment]
Please have a look here: https://de.slideshare.net/holgerschill/using-xcore-with-xtext

**Comment by holgerschill** [comment]
And here: https://github.com/holgerschill/XtextCon_Xcore_Maven

**Comment by cdietrich** [comment]
[maven-xcore-sample.zip](https://github.com/eclipse/xtext-maven/files/2069502/maven-xcore-sample.zip) sample project

**Comment by cdietrich** [comment]
Fixed in 2.15
